python/ray/serve/_private/replica.py

```diff
@@ -331,26 +341,51 @@ def record_request_metrics(self, *, route: str, latency_ms: float, was_error: bo
     def _push_autoscaling_metrics(self) -> Dict[str, Any]:
         look_back_period = self._autoscaling_config.look_back_period_s
         self._metrics_store.prune_keys_and_compact_data(time.time() - look_back_period)
+
+        new_aggregated_metrics = {}
+        new_metrics = {**self._metrics_store.data}
+
+        if not RAY_SERVE_COLLECT_AUTOSCALING_METRICS_ON_HANDLE:
+            # Keep the legacy window_avg ongoing requests in the merged metrics dict
+            window_avg = self._metrics_store.aggregate_avg([self._replica_id])[0] or 0.0
+            new_aggregated_metrics.update({RUNNING_REQUESTS_KEY: window_avg})
+
         replica_metric_report = ReplicaMetricReport(
             replica_id=self._replica_id,
             timestamp=time.time(),
-            aggregated_metrics={
-                RUNNING_REQUESTS_KEY: self._metrics_store.aggregate_avg(
-                    [self._replica_id]
-                )[0]
-                or 0.0
-            },
-            metrics={
-                RUNNING_REQUESTS_KEY: self._metrics_store.data.get(self._replica_id, [])
-            },
+            aggregated_metrics=new_aggregated_metrics,
+            metrics=new_metrics,
         )
         self._controller_handle.record_autoscaling_metrics_from_replica.remote(
             replica_metric_report
         )
 
-    def _add_autoscaling_metrics_point(self) -> None:
+    def _replica_ongoing_requests(self) -> Dict[ReplicaID, int]:
+        return {self._replica_id: self._num_ongoing_requests}
```

Comment:
maybe we can change they key here to be `RUNNING_REQUESTS_KEY` instead of `replica_id`